src/components/form.js

```diff
@@ -1,14 +1,50 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { AddBook } from '../redux/book/book';
 
 export default function Form() {
+  const [book, setBook] = useState({
+    title: '',
+    author: '',
+    id: Math.floor(Math.random() * 100000),
+  });
+  const { title, author } = book;
+  const dispatch = useDispatch();
+  const handleChange = (e) => {
+    setBook((prevState) => ({
+      ...prevState,
+      [e.target.name]: e.target.value,
+    }));
+  };
+
+  const addBookSubmit = (e) => {
+    e.preventDefault();
+    if (book.title !== '' && book.author !== '') {
+      dispatch(AddBook(book));
```

Comment:
- [OPTIONAL] Please ensure a unique Id is generated each time a new book is created, and also make sure the inputs are cleared after submitting the form.